diff --git a/slsh/Makefile.in b/slsh/Makefile.in
index cba9d81..cb82774 100644
--- a/slsh/Makefile.in
+++ b/slsh/Makefile.in
@@ -32,6 +32,7 @@ MKINSDIR	= $(CONFIG_DIR)/autoconf/mkinsdir.sh
 # Where system-wide slsh.rc, scripts, and library files reside
 #----------------------------------------------------------------------------
 SLSH_CONF_DIR	= @sysconfdir@
+SLSH_DEF_DIR	= $(datarootdir)/slsh
 SLSH_LIB_DIR	= $(datarootdir)/slsh
 SLSH_HELP_DIR	= $(SLSH_LIB_DIR)/help
 SLSH_DOC_DIR	= $(datarootdir)/doc/slsh
@@ -60,7 +61,7 @@ MAN_DIR		= @mandir@/man1
 DEST_LIB_DIR	= -L$(DESTDIR)@libdir@
 DEST_BIN_DIR	= $(DESTDIR)$(BIN_DIR)
 DEST_MAN_DIR	= $(DESTDIR)$(MAN_DIR)
-DEST_SLSH_CONF_DIR = $(DESTDIR)$(SLSH_CONF_DIR)
+DEST_SLSH_CONF_DIR = $(DESTDIR)$(SLSH_DEF_DIR)
 DEST_SLSH_LIB_DIR  = $(DESTDIR)$(SLSH_LIB_DIR)
 DEST_SLSH_HELP_DIR = $(DESTDIR)$(SLSH_HELP_DIR)
 DEST_SLSH_LOCALLIB_DIR = $(DESTDIR)$(SLSH_LOCALLIB_DIR)
@@ -80,7 +81,7 @@ SHELL = /bin/sh
 INST_LIBS = $(DEST_LIB_DIR) $(RPATH) $(SLANG_INST_LIB) -lslang $(READLINE_LIB) $(DYNAMIC_LIBS)
 DEFS = -DSLSH_CONF_DIR='"$(SLSH_CONF_DIR)"' -DSLSH_PATH='"$(SLSH_LIB_DIR)"' \
  -DSLSH_CONF_DIR_ENV='$(SLSH_CONF_DIR_ENV)' -DSLSH_LIB_DIR_ENV='$(SLSH_LIB_DIR_ENV)' \
- -DSLSH_PATH_ENV='$(SLSH_PATH_ENV)' $(SLSYSWRAP_DEF)
+ -DSLSH_PATH_ENV='$(SLSH_PATH_ENV)' -DSLSH_DEF_DIR='"$(SLSH_DEF_DIR)"' $(SLSYSWRAP_DEF)
 SDEFS = $(DEFS) -DSLSH_STATIC
 SRC_LIBS = $(SLANG_ELFLIB) -lslang $(SLSYSWRAP_LIB) $(READLINE_LIB) $(DYNAMIC_LIBS)
 STATIC_SRC_LIBS = $(SLANG_OBJLIB) -lslang $(SLSYSWRAP_LIB) $(READLINE_LIB) $(STATIC_LIBS)
diff --git a/slsh/slsh.c b/slsh/slsh.c
index 623c94a..12cbaf9 100644
--- a/slsh/slsh.c
+++ b/slsh/slsh.c
@@ -372,7 +372,7 @@ static int load_startup_file (const char *pgm, int is_interactive)
    if (NULL == dir)
      {
 #ifdef SLSH_CONF_DIR
-	dir = SLSH_CONF_DIR;
+	dir = SLSH_CONF_DIR ":" SLSH_DEF_DIR;
 	if (dir != NULL)
 	  {
 	     status = try_to_load_file (dir, (char *)SLSHRC_FILE, NULL);
